src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/deformable_convolution_gpu_bfyx_ref.cl

```diff
@@ -48,85 +48,90 @@ KERNEL(deformable_convolution_gpu_bfyx_ref)(
     const uint in_split_offset = conv_group * INPUT0_FEATURE_PITCH * FILTER_IFM_NUM;
 #endif
 
-    for (uint dg = 0; dg < DEFORMABLE_GROUPS; ++dg)
-    {
-        const int ifm_offset = dg * channel_per_deformable_group;
-        const int dg_offset = dg * FILTER_SIZE_Y * FILTER_SIZE_X * OUTPUT_SIZE_Y * OUTPUT_SIZE_X;
-        const int trans_offset = b * INPUT1_BATCH_PITCH + 2 * dg_offset;
-#if DEFORMABLE_MASK_ENABLED
-        const int mask_offset = b * INPUT2_BATCH_PITCH + dg_offset;
+    const int offset_size = FILTER_SIZE_Y * FILTER_SIZE_X * OUTPUT_SIZE_Y * OUTPUT_SIZE_X;
+
+    for (uint c = 0; c < FILTER_IFM_NUM; ++c) {
+#if GROUPED
+        const int deformable_group_idx = (FILTER_IFM_NUM * of + c) / (FILTER_IFM_NUM * FILTER_GROUPS_NUM / DEFORMABLE_GROUPS);
+#else
+        const int deformable_group_idx = (FILTER_IFM_NUM * of + c) / (FILTER_IFM_NUM * FILTER_GROUPS_NUM / DEFORMABLE_GROUPS) % DEFORMABLE_GROUPS;
```

Comment:
I am not familiar with this kernel but original impl was handling batch in the offset while the new impl is not. Could you explain whats the difference?

Reply:
It is implemented by referring to `src/core/reference/include/ngraph/runtime/reference/deformable_convolution.hpp` and batch is not used in it to calculate `deformable_group_idx`. So the batch is not used in this code.

Reply:
Added test cases of batch = 2, 3, 4 to group/deformable_group = 2/2, 4/1, 4/2.